### PR TITLE
882 switch start method in parallel executor

### PR DESCRIPTION
**Pull request by yafimvo**
## Describe your changes
Code fixed, test for default value, and documentation added.

Please note that versionadded:: X.Y.Z didn't set.

## Issue ticket number and link
Closes #https://github.com/ploomber/ploomber/issues/942 

## Checklist before requesting a review
- [ X] I have performed a self-review of my code
- [ X] I have added thorough tests (when necessary).
- [ X] I have added the right documentation (when needed). Product update? If yes, write one line about this update.

### CODE REVIEW

File: src/ploomber/executors/parallel.py [hunk 1]
```diff
@@ -93,13 +104,25 @@ class Parallel(Executor):
 
     # NOTE: Tasks should not create child processes
     # https://docs.python.org/3/library/multiprocessing.html#multiprocessing.Process.daemon
+    multiprocessing_start_methods = ['fork', 'spawn', 'forkserver']
 
-    def __init__(self, processes=None, print_progress=False):
+    def __init__(self,
```
Comment (idomic) [inline src/ploomber/executors/parallel.py:109]: Why is the code showing as changed? We already merged those.

**Review by idomic — CHANGES_REQUESTED** [review]
Commented

**Review by idomic — CHANGES_REQUESTED** [review]
Looks good, please add an updated docstring into the changelog + doc updates into the API docs doc/api/spec.rst

File: CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +10,7 @@
 * `env.yaml` allows to use existing keys in subsequent values
 * Fixes missing plot when generating `ploomber report` with `--backend d3`
 * Fixes error when using dotted paths in `grid` (#951)
+* Add `start_method` to Parallel executor to set the method which should be used to start child processes ([#942](https://github.com/ploomber/ploomber/issues/942))
```
Comment (idomic) [inline CHANGELOG.md:13]: Copy the syntax from the line before. Also pull from main, 0.20 was already released.

Comment (idomic) [reply CHANGELOG.md:13]: Also, you need to add a line on the docs folder, this was the initial issue ( we were missing documentation on this feature).

**Review by idomic — CHANGES_REQUESTED** [review]
Add fixes

File: CHANGELOG.md (hunk 2 above)
Comment (yafimvo) [reply CHANGELOG.md:12]: What do you mean by adding a line on the docs folder?
I've added a description of this new argument (with a note) to `src/ploomber/executors/parallel.py` next to other similar arguments (`processes` and `print_progress`). I saw the [spec.rst](https://github.com/ploomber/ploomber/blob/master/doc/api/spec.rst#executor) but it seems like it's not the place to document new arguments.

image

**Comment by yafimvo** [comment]
I ran a local test on 'tests/products/test_file.py::test_task_with_client_and_metaproduct_with_some_missing_products' and I'm getting this error:

No module named 'ploomber.papermill.engine'

I think there are missing files in papermill directory (my papermill dir is empty as well as on master branch after pull)

**Comment by idomic** [comment]
Try pulling from Main and redo the conda env.

**Comment by idomic** [comment]
On the docs, you'd need to add a line specifying the change here: doc/api/spec.rst

**Comment by idomic** [comment]
On the docs, you'd need to add a line specifying the change here: doc/api/spec.rst

**Review by idomic — APPROVED** [review]
lgtm, nice job!

**Comment by idomic** [comment]
@yafimvo solve conflicts before we can merge.

**Comment by idomic** [comment]
Closes https://github.com/ploomber/ploomber/issues/942

**Pull request merged**